tests/unit/components/playback/controls/AddUser.spec.js

```diff
@@ -0,0 +1,371 @@
+    ],
+    ["", "<empty>", "alphanumeric-id", "validate_uuidBase_fiftyseven_encode"],
+    ["Experiment anemia -1", "valid input", "nickname-id", "validate_nickname"],
+    ["Cat * lab", "contains asterisk *", "nickname-id", "validate_nickname"],
+    ["Cat lab`", "contains left quote `", "nickname-id", "validate_nickname"],
+    ["Cat lab;", "contains semi-colon ;", "nickname-id", "validate_nickname"],
+    [
+      "Experiment anemia alpha cells -1",
+      "more than 20 characters",
+      "nickname-id",
+      "validate_nickname",
+    ],
+    ["C", "minimum one character C", "nickname-id", "validate_nickname"],
+    ["", "<empty>", "nickname-id", "validate_nickname"],
+  ])(
+    "When the text %s (%s) is entered into the field found with the selector ID %s, Then the correct text validation function (%s) is called and the error message from the validation function is rendered below the input in the DOM",
+    async (
+      entry,
+      test_description,
+      selector_id_suffix,
+      text_validation_type
+    ) => {
+      const spied_text_validator = jest.spyOn(
+        TextValidation.prototype,
+        text_validation_type
+      );
+
+      const target_input_field = wrapper.find(
+        "#input-widget-field-" + selector_id_suffix
+      );
+
+      const target_error_message = wrapper.find(
+        "#input-widget-feedback-" + selector_id_suffix
+      );
+
+      target_input_field.setValue(entry);
+
+      await Vue.nextTick();
+      expect(spied_text_validator).toHaveBeenCalledWith(entry);
+
+      expect(target_error_message.text()).toStrictEqual(
+        spied_text_validator.mock.results[0].value
+      );
+    }
+  );
+});
+
+describe("AddUser.enable_save_button", () => {
+  beforeEach(async () => {
+    const propsData = {
+      dialogdata: null,
+      dataindex: 0,
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    store = await NuxtStore.createStore();
+  });
+
+  beforeAll(async () => {
+    // note the store will mutate across tests, so make sure to re-create it in beforeEach
+    const storePath = `${process.env.buildDir}/store.js`;
+    NuxtStore = await import(storePath);
+  });
+
+  afterEach(() => wrapper.destroy());
+
+  test.each([
+    [
+      "0VSckkBYH2An3dqHEyfRRE",
+      "Experiment anemia -1",
+      "The entered ID has an invalid character 0,",
+      "",
+      "color: rgb(63, 63, 63);",
+    ],
+    [
+      "5FY8KwTsQaUJ2KzHJGetfE",
+      "Cat * lab",
+      "",
+      "Invalid character present. Valid characters are alphanumeric & # - . _  ( ) /",
+      "color: rgb(63, 63, 63);",
+    ],
+    [
+      "5FY8KwTsQaUJ2KzHJGetfE",
+      "Experiment anemia -1",
+      "",
+      "",
+      "color: rgb(255, 255, 255);",
+    ],
+  ])(
+    "Given an UUID (%s),  Nickname (%s) for 'Add User' as input, When the input contains based on validthe critera or failure (%s)(%s)(%s), Then display of Label 'Save ID' is visible or greyed (%s)",
```

Comment:
testing the error messages is already done in another test.  This one should just focus on whether the Save button is enabled or disabled based on the input.  Please also update the test name to reflect this.  And apply this comment to all 4 components

Reply:
Agreed.. and corrected.